functions/go/ensure-name-segment/ensure_name_segment.go

```diff
@@ -0,0 +1,203 @@
+package main
+
+import (
+	"fmt"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/kustomize/api/filters/prefixsuffix"
+	"sigs.k8s.io/kustomize/api/resid"
+	"sigs.k8s.io/kustomize/api/resmap"
+	"sigs.k8s.io/kustomize/api/resource"
+	"sigs.k8s.io/kustomize/api/types"
+	"sigs.k8s.io/kustomize/kyaml/yaml"
+	k8syaml "sigs.k8s.io/yaml"
+)
+
+const (
+	fnConfigGroup      = "fn.kpt.dev"
+	fnConfigVersion    = "v1alpha1"
+	fnConfigAPIVersion = fnConfigGroup + "/" + fnConfigVersion
+	fnConfigKind       = "EnsureNameSegment"
+)
+
+type EnsureNameSegment struct {
+	yaml.ResourceMeta `json:",inline" yaml:",inline"`
+	// Segment is the desired name segment.
+	Segment string `json:"segment" yaml:"segment"`
+	// ActionOnNotFound controls the desired action when the desired segment is not found in the name.
+	// If not specified, prepend will be the default.
+	ActionOnNotFound ActionMode `json:"actionOnNotFound,omitempty" yaml:"actionOnNotFound,omitempty"`
```

Comment:
`EditMode` might be more intuitive.

Reply:
I'm a little confused here. Are you suggesting  changing `actionOnNotFound` to `EditMode` or change `ActionMode`?

Reply:
oh.. change `actionOnNotFound` to `editMode`, the user-facing part.